src/server/HttpRequest.ts

```diff
@@ -1,6 +1,7 @@
 import type { IncomingMessage } from 'http';
+import type { Guarded } from '../util/GuardedStream';
 
 /**
  * An incoming HTTP request;
  */
-export type HttpRequest = IncomingMessage;
+export type HttpRequest = Guarded<IncomingMessage>;
```

Comment:
Wouldn't that cause problems? Because it's not what Node returns?

Reply:
This forces the HttpServer to guard the stream that it gets from Node before it can call the LDP handler which happens [here](https://github.com/solid/community-server/pull/327/files#diff-d5a884e2ac05dfa4387ece575e04484ba6cc961b355a48a9aa66e848f14d918aR44).